src/OrchardCore.Modules/OrchardCore.Settings/SiteSettings.cs

```diff
@@ -1,11 +1,22 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
 using Microsoft.AspNetCore.Routing;
+using Newtonsoft.Json.Linq;
 using OrchardCore.Documents;
+using OrchardCore.Entities;
 
 namespace OrchardCore.Settings
 {
     // When updating class also update SiteSettingsDeploymentSource and SettingsStep.
-    public class SiteSettings : DocumentEntity, ISite
+    public class SiteSettings : DocumentEntity, ICacheableEntity, ISite
     {
+        private static readonly JObject _defaultProperties = new();
+
+        private readonly ConcurrentDictionary<string, object> _cache = new();
```

Comment:
I don't think this is needed. SiteSettings is resolved by using `ISiteService` ISiteService access data from the database so it is not thread-safe and you should always await it. and `we do not have `.AsAsync<>`